src/main/java/de/tum/in/www1/artemis/service/connectors/localci/buildagent/BuildAgentSSHKeyService.java

```diff
@@ -0,0 +1,110 @@
+package de.tum.in.www1.artemis.service.connectors.localci.buildagent;
+
+import static de.tum.in.www1.artemis.config.Constants.PROFILE_BUILDAGENT;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.attribute.PosixFilePermissions;
+import java.security.GeneralSecurityException;
+import java.security.KeyPair;
+import java.security.KeyPairGenerator;
+import java.util.Optional;
+
+import org.apache.sshd.common.config.keys.writer.openssh.OpenSSHKeyEncryptionContext;
+import org.apache.sshd.common.config.keys.writer.openssh.OpenSSHKeyPairResourceWriter;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.boot.context.event.ApplicationReadyEvent;
+import org.springframework.context.annotation.Profile;
+import org.springframework.context.event.EventListener;
+import org.springframework.stereotype.Service;
+
+@Service
+@Profile(PROFILE_BUILDAGENT)
+public class BuildAgentSSHKeyService {
+
+    private static final Logger log = LoggerFactory.getLogger(BuildAgentSSHKeyService.class);
+
+    private KeyPair keyPair;
+
+    @Value("${artemis.version-control.ssh-private-key-folder-path:#{null}}")
```

Comment:
Considering these keys are supposed to be ephemeral, would it not make sense to default to a temp folder?
Setting this to a specific folder would likely only be needed in edge-cases.

Reply:
Defining a default value here isn't a good idea. This folder contains the private SSH key, so it needs to be kept highly confidential with proper access controls and permissions.

If we provide a default value, an inexperienced Artemis admin might use it without considering the security implications, which could pose a risk. It's better to require each admin to choose a secure location for storing SSH keys and configure the path themselves.